server.js

```diff
@@ -33,139 +21,33 @@ parser.addArgument('--verbose', {action: "storeTrue", help: "verbose server logg
-app.route("/dist/*") // Auspice hardcodes /dist/… in its Webpack config.
-  .all(expressStaticGzip(auspiceAssetPath(), {maxAge: '30d'}));
-
-
-/* Charon API used by Auspice.
- */
-app.routeAsync("/charon/getAvailable")
-  .all(cors({origin: 'http://localhost:8000'})) // allow cross-origin from the gatsby dev server
-  .getAsync(auspiceServerHandlers.getAvailable);
-
-app.routeAsync("/charon/getDataset")
-  .getAsync(auspiceServerHandlers.getDataset);
-
-app.routeAsync("/charon/getNarrative")
-  .getAsync(auspiceServerHandlers.getNarrative);
-
-app.routeAsync("/charon/getSourceInfo")
-  .getAsync(auspiceServerHandlers.getSourceInfo);
-
-app.routeAsync("/charon/*")
-  .all((req) => {
-    utils.warn(`(${req.method}) ${req.url} has not been handled / has no handler`);
-    throw new NotFound();
-  });
-
-/* Specific routes to be handled by Gatsby client-side routing
- */
-app.route([
-  "/users/:user",
-  "/groups",
-  "/groups/:groupName"
-]).get((req, res) => res.sendFile(gatsbyAssetPath("index.html")));
-
-/* Routes which are plausibly for auspice, as they have specific path signatures.
- * We verify if these are are backed by datasets and hand to auspice if so.
- */
-app.route(potentialAuspiceRoutes).get(
-  isRequestBackedByAuspiceDataset,
-  sendAuspiceHandler,
-  sendGatsbyHandler
-);
-
-/**
- * Finally, we catch all remaining routes and send to Gatsby
- */
-app.get("*", sendGatsbyHandler);
-
-
-const server = app.listen(app.get('port'), () => {
-  console.log("  -------------------------------------------------------------------------");
-  console.log(nextstrainAbout);
-  console.log(`  Server listening on port ${server.address().port}`);
-  console.log(`  Accessible at https://nextstrain.org or http://localhost:${server.address().port}`);
-  console.log(`  Server is running in ${production ? 'production' : 'development'} mode`);
-  console.log("\n  -------------------------------------------------------------------------\n\n");
-}).on('error', (err) => {
-  if (err.code === 'EADDRINUSE') {
-    utils.error(`Port ${app.get('port')} is currently in use by another program.
-    You must either close that program or specify a different port by setting the shell variable
-    "$PORT". Note that on MacOS / Linux, "lsof -n -i :${app.get('port')} | grep LISTEN" should
-    identify the process currently using the port.`);
-  }
-  utils.error(`Uncaught error in app.listen(). Code: ${err.code}`);
-});
-
-
-function sendGatsbyHandler(req, res) {
-  utils.verbose(`Sending Gatsby entrypoint for ${req.originalUrl}`);
-  return res.sendFile(
-    gatsbyAssetPath(""),
-    {},
-    (err) => {
-      // callback runs when the transfer is complete or when an error occurs
-      if (err) res.status(404).sendFile(gatsbyAssetPath("404.html"));
+const port = process.env.PORT || 5000;
+
+/* Import app after setting global.verbose so that calls to utils.verbose()
+ * respect our --verbose option as expected.
+ */
+const app = require("./src/app");
```

Comment:
[no changes requested] Could you comment on the decision to use this form, rather than creating a function in `./src/app.js` which you'd call here along the lines of `const app = setupExpressServer()`? This construction feels strange to me, but it works and it seems like lots of people use it.

Reply:
Sure. IIUC, you're asking why I used

```
const app = require("./src/app");
```

instead of something like:

```
const {setupExpressApp} = require("./src/app");
const app = setupExpressApp();
```

If so, the basic reason is that it was less churn/change in the `server.js` → `src/app.js` move to keep the existing `app` variable and simply export it than to wrap everything in a function, and thus the diff is easier to read.

Functionally, the two forms are ~identical; the difference is the extra level of indirection that the second form provides, which would definitely be warranted if we ever have a need to instantiate multiple copies of the Express app object. Without that need right now, I kept it slightly simpler.